go/libraries/doltcore/sqle/index/prolly_fields.go

```diff
@@ -22,12 +22,62 @@ import (
 	"github.com/dolthub/go-mysql-server/sql"
 	"github.com/shopspring/decimal"
 
-	"github.com/dolthub/dolt/go/libraries/doltcore/schema/typeinfo"
 	geo "github.com/dolthub/dolt/go/store/geometry"
-	"github.com/dolthub/dolt/go/store/types"
 	"github.com/dolthub/dolt/go/store/val"
 )
 
+// todo(andy): this should go in GMS
+func DenormalizeRow(sch sql.Schema, row sql.Row) (sql.Row, error) {
```

Comment:
These feel like the worst things ever but I guess it's okay for now. GMS should absolutely be taking care of this.

Reply:
Yeah I agree. We need another pass to move this logic to GMS